amf-api-contract/shared/src/main/scala/amf/apicontract/internal/validation/shacl/graphql/GraphQLObjects.scala

```diff
@@ -20,34 +20,55 @@ case class GraphQLObject(node: NodeShape) {
 
   def inherits: Seq[GraphQLObject] = node.inherits.map(_.asInstanceOf[NodeShape]).map(GraphQLObject)
 
+  def isValidInputType: Boolean = {
```

Comment:
You should check the `isInput` field in Objects

Reply:
that's what's done in the `else` block